torchci/pages/api/flaky-tests/flakytest.ts

```diff
@@ -18,5 +23,24 @@ export default async function handler(
     file as string
   );
 
-  res.status(200).json({ flakyTests });
+  // capture looks like: testName, testSuite
+  const flakySamples: {
+    [capture: string]: JobData[];
+  } = {};
+
+  // only get log view for fewer than 5 flaky tests (to not spam query calls), which should be every case as we
+  // now limit on test name
+  console.debug(`Retrieved ${flakyTests.length} flaky tests`);
+  if (flakyTests.length < 5) {
+    const unfulfilledPromises = flakyTests.map(async function(flakyTest) {
+      return await fetchFailureSamples(getFlakyTestCapture(flakyTest));
```

Comment:
if we're awaiting them in here, then they aren't really unfufilled; they actually get executed. I think you can get rid of the async part on line 35 and remove the await on here.